src/figuresClasses.ts

```diff
@@ -1,11 +1,90 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public readonly shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('The sides must be greater than 0');
+    }
+
+    const longestSide = Math.max(a, b, c);
+
+    if (longestSide >= a + b + c - longestSide) {
+      throw new Error('The longest side must be less than the others');
+    }
+  }
+
+  getArea(): number {
+    const { a, b, c } = this;
+
+    const p = (a + b + c) / 2;
+
+    const s =
+      Math.floor(Math.sqrt(p * (p - a) * (p - b) * (p - c)) * 100) / 100;
+
+    return s;
+  }
+}
+
+export class Circle implements Figure {
+  public readonly shape: Shape = 'circle';
 
-export class Circle implements Figure {}
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    if (this.radius <= 0) {
+      throw new Error("This circle doesn't exist");
+    }
+  }
+
+  getArea(): number {
+    const { radius } = this;
```

Comment:
i think it is not a good idea destructuring this keyword. It is non obvious
Especially here when you need to get only one property